pkgs/development/python-modules/cachier/default.nix

```
@@ -0,0 +1,80 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, setuptools
+, watchdog
+, portalocker
+, pathtools
+, pytestCheckHook
+, pymongo
+, dnspython
+, pymongo-inmemory
+, pandas
+}:
+
+buildPythonPackage rec {
+  pname = "cachier";
+  version = "2.2.1";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-nm98LT87Z7yErKvIqMp93OEX9TDojqqtItgryHgSQJQ=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    watchdog
+    portalocker
+    pathtools
+  ];
+
+  preCheck = ''
+    substituteInPlace pytest.ini \
+      --replace  \
+        "--cov" \
+        "#--cov"
+  '';
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pymongo
+    dnspython
+    pymongo-inmemory
+    pandas
+  ];
+
+  disabledTests = [
+    # touches network
+    "test_mongetter_default_param"
+    "test_stale_after_applies_dynamically"
+    "test_next_time_applies_dynamically"
+    "test_wait_for_calc_"
+    "test_precache_value"
+    "test_ignore_self_in_methods"
+    "test_mongo_index_creation"
+    "test_mongo_core"
+
+    # nah
```

```suggestion
    # don't test formatting
```
or something like that